Update Access Rules

VincentsSin opened the pull request:
Add owl statues to region logic

VincentsSin committed:
Update regions.lua

Add owl statues to region logic

VincentsSin committed:
Update regions.lua

Fixed grouping for Snowhead region access.

VincentsSin committed:
Update regions.lua

VincentsSin commented:
Added "$clear_snowhead" to springtime goron shop

[snowhead.jsonc](https://github.com/user-attachments/files/24268949/snowhead.jsonc)

G4M3RL1F3 committed:
Simplified and corrected some region rules

More work is needed to match the 1.0 logic.

G4M3RL1F3 commented:
I will note that the logic must match whatever is in [NormalRules.py in the 1.0 apworld](https://github.com/RecompRando/Archipelago-MM/blob/better-logic-literally-all-the-things-with-0-9-5/worlds/mm_recomp/NormalRules.py). I will give you that some things in the tracker files are not setup properly to make it match the apworld such as a few more region rules, like upper and lower Ikana Canyon that need to be separated.

VincentsSin commented:
I removed the zipped file so I can go through and make sure it matches. I'll re-upload when I've confirmed

G4M3RL1F3 committed:
Added can_warp_out macro

It was added in regions.lua, so adding it here as needed to avoid issues.

VincentsSin committed:
Merge branch 'G4M3RL1F3:giga_sanity_rando' into giga_sanity_rando

VincentsSin committed:
Added Souls To Existing Logic

VincentsSin committed:
Update layouts.lua

Add enemy souls to sanity tab

VincentsSin committed:
Add Enemy Soul Tracking

VincentsSin committed:
Mapped Enemy Souls

VincentsSin committed:
Create test

VincentsSin committed:
Add files via upload

VincentsSin committed:
Delete images/pack_settings/sanities/enemies/test

VincentsSin committed:
Add Enemy Souls

VincentsSin committed:
Update Zora Souls to Match APworld

G4M3RL1F3 committed:
Added "Souls" tab for the item tracker
I want to make sure that people with smaller screens can fit all items within the width.

G4M3RL1F3 committed:
Added boss souls

G4M3RL1F3 committed:
Fixed pot game & hall zora and coast zora souls not displaying on the item tracker

G4M3RL1F3 committed:
Removed soul of coast zora
It went from a progression item to a useful one, making it redundant to keep.

G4M3RL1F3 committed:
Simplified key sanity off options handling

G4M3RL1F3 committed:
Added miscellaneous souls
Will need to return later for butterflies when they're implemented.

G4M3RL1F3 committed:
Updated some sprites

G4M3RL1F3 committed:
Added scarecrow sanity
When adding logic to locations, make duplicates of existing access rules (for normal rules) and add these:
- For original: $scarecrow_sanity_off
- For duplicated: scarecrow_sanity
Make sure to add ocarina if they're not present.

VincentsSin committed:
Add files via upload

VincentsSin committed:
Add files via upload

G4M3RL1F3 committed:
Fixed shop related and autotracking issues

G4M3RL1F3 committed:
Added absurd sanity soul option

G4M3RL1F3 committed:
Added NPC Souls option

VincentsSin committed:
Add Addresses To Location Mapping

The first entry of a location group will have the name notated, with everything under as part of that group.

VincentsSin committed:
Add Grass Locations To Tracker

I may have overlooked a logic situation, but at the very least they account for absurd souls being enabled or not.

VincentsSin committed:
Add Location References To Grass Addresses

VincentsSin committed:
Add Grass To Secret Shrine

Also updated region logic to split Ikana into upper and lower regions.

VincentsSin committed:
Update Ikana To Upper/Lower Regions

VincentsSin committed:
Add Macros For Owls & Warp-Out

Also added some for frogs, but may scrap it.

VincentsSin committed:
Add Start Of Moon-Mask Requirements

VincentsSin committed:
Add Pots To Tracker

VincentsSin committed:
Add Scarecrow Logic

VincentsSin committed:
Add Pot Locations to Matching Addresses

VincentsSin committed:
Add files via upload

G4M3RL1F3 committed:
Cleaned-up existing scarecrow sanity rules

G4M3RL1F3 committed:
Cleaned-up existing NPC soul rules

G4M3RL1F3 committed:
Fixed location IDs duplicates

G4M3RL1F3 committed:
Cleaned-up existing absurd soul rules

G4M3RL1F3 committed:
Cleaned-up existing misc. soul rules

G4M3RL1F3 committed:
Updated enemy soul sprites and related cleanup

G4M3RL1F3 committed:
Image folders reorganization

G4M3RL1F3 committed:
Added enemy soul option and updated related rules

VincentsSin committed:
Add Hitspots, Invisible & Freestanding Rupees

VincentsSin committed:
Add Rupee Locations To Addresses

VincentsSin committed:
Add files via upload

VincentsSin committed:
Matching Upstream Commits

VincentsSin committed:
Upload To Match PR Commits

VincentsSin committed:
Add files via upload

VincentsSin committed:
Update clock_town.jsonc

Fixed incorrect absurd soul rule

VincentsSin committed:
Update dungeons.jsonc

Fix missed access rule

VincentsSin committed:
Update great_bay.jsonc

Fix access rule

VincentsSin committed:
Update location_mapping.lua

G4M3RL1F3 committed:
Added frogsanity and simplified rules, also cleaned some stuff

VincentsSin committed:
Added Rocks, Soil, Snowballs, and Crates

VincentsSin committed:
Icons F/ Crates, Snowballs, & Soil

VincentsSin committed:
Add Locations To Addresses 

Crates/Snowballs/Soil/Rocks

VincentsSin committed:
Update location_mapping.lua

Small fix to some barrels in Pirates' Fortress

G4M3RL1F3 committed:
Added grass sanity option

G4M3RL1F3 committed:
Aligned the Stone Tower squares on the map
It was triggering my OCD

G4M3RL1F3 committed:
Cleaned-up "Start with consumables" images
Another OCD triggering thing. There was leftover black backdrop from another setting that persisted in this one here. It is now fixed.

G4M3RL1F3 committed:
Re-organized settings popup window

G4M3RL1F3 committed:
Simplified handlings for playing songs and soaring to owls

G4M3RL1F3 committed:
Simplified bottle count handling

G4M3RL1F3 committed:
Touched missed parts for bottle count handling

G4M3RL1F3 committed:
Fixed spelling mistakes and forgotten function changes that were causing logic issues

G4M3RL1F3 committed:
Simplified remain count handling and fixed some logic errors
Also added better boss soul handling because I somehow forgot about it.

G4M3RL1F3 committed:
Added pot sanity option

G4M3RL1F3 committed:
Added mask requirements for moon and Majora's Lair access

VincentsSin committed:
Sanity icons

VincentsSin committed:
Add remaining sanities to tracker

VincentsSin committed:
Added remaining sanity locations to addresses

VincentsSin commented:
~~Not sure why, but some locations won't show in logic even with everything shown as obtained.~~ Re-verified files and it was logic.lua that didn't copy over properly.

VincentsSin committed:
Add small snowball access rules

VincentsSin committed:
Small snowballs again...

Realized one is past the large snowball barricade, requiring different rules

G4M3RL1F3 committed:
Fixed forever red checks

VincentsSin committed:
Add Star Fox Requirements

Didn't add images, I'm unsure of font style

VincentsSin committed:
Add Frog & Scarecrow Requirements

Again no images

VincentsSin commented:
Owl requirements were also added in last commit

VincentsSin committed:
Add Visibility Rules To Sanity Locations

No images to assist, but made my own locally to ensure they work.

VincentsSin committed:
Adjust Real Fairysanity Locations

Added Song of Storms as viable method

G4M3RL1F3 committed:
Fixed Termina Field Invisible Rupee Over Water not tracking

VincentsSin committed:
Fix gossip fairy tracking near Takkuri

G4M3RL1F3 committed:
Attempted fix at groups clearing even when clear_as_group = false

VincentsSin committed:
Arrange slot data options to alphabetical order

VincentsSin committed:
Uploaded wrong lua file

G4M3RL1F3 committed:
Fixed options not auto-populating and some locations not auto-tracking
There are more locations that still don't auto-track.

VincentsSin commented:
I've been going through a test async to find specifically which locations have not been tracking, and making corrections that have, so far, fixed what I've found. Most were uncaught location name changes/spelling errors, but a select few I had the wrong addresses copied over. I should have everything updated by the end of the week.

VincentsSin committed:
Various corrections and updates

VincentsSin committed:
Placeholder images for most new settings

G4M3RL1F3 committed:
Fixed autotracking for 6am songwall and guru guru, and some ID duplicates

VincentsSin committed:
Corrected meeting postman notebook location

VincentsSin committed:
Fix Gorman Brother's checks

VincentsSin committed:
Fix location description for Rock Grotto Chest

Updated location mapping for Ikana Road to Ikana Rock Grotto.

VincentsSin committed:
Fix location mapping for North Clock Town West Tree

G4M3RL1F3 committed:
Changed pack selection list name displayed
Changed from "Majoras Mask AP" to "Majora's Mask Recompiled". I just wanted the name to represent the Recomp rando version. The previous one made it a bit ambiguous as to which randomizer it was for.

G4M3RL1F3 committed:
Fixed Deku Playground rupees not tracking

G4M3RL1F3 committed:
Reduced item_count for WFT Dragonfly Room flowers from 4 to 3

G4M3RL1F3 committed:
Fixed name mismatch for WFT Main Room flower and removed item_count

G4M3RL1F3 committed:
Removed item_count number display if they were set to 1
It is more conclusive to look at a check without a number rather than having the number 1 displayed. If the number 1 is displayed, a player might think that all but one location was checked for that "group", potentially bringing some confusion in the player. It is a very insignificant change, but knowing myself, my brain would start a train of thoughts just because the number 1 is displayed.

VincentsSin committed:
Various changes; mostly corrections

G4M3RL1F3 committed:
Fixed shop locations lowering the price by 1 when checked instead of full clearing
I also disabled shop prices handling. The slot data info used to fetch prices has been removed from the apworld for some unknown reason. It was breaking the auto-population of settings.

G4M3RL1F3 committed:
Removed unused inverse rules

G4M3RL1F3 committed:
Fixed boss warps and star fox settings not auto-populating

G4M3RL1F3 committed:
Changed Boss Warps with Remains fix to match main branch

VincentsSin committed:
Various minor updates

VincentsSin committed:
Fix accidental reversion of boss remains settings

VincentsSin committed:
Corrected snowball tracking in path to village

VincentsSin committed:
Missed the upper snowball when correcting

G4M3RL1F3 committed:
Updated location name for midnight meeting to reflect the times it can be checked
Plus some minor cleanup

VincentsSin committed:
Added Termina Field sub-map

G4M3RL1F3 committed:
Moved Termina Field map into Central tab and removed location size overrides

G4M3RL1F3 committed:
Updated IDs and implemented map switching
Map switching could not be tested as of this PR: might need to return to this.

G4M3RL1F3 committed:
Functional auto tab switching

G4M3RL1F3 committed:
Added Zora Hall map

G4M3RL1F3 committed:
Separated Zora Hall scarecrow to have it's own square

G4M3RL1F3 committed:
Several name mismatches and missing locations fixes

G4M3RL1F3 commented:
Don't know what happened. Seems like the PR closed on its own after a commit??

G4M3RL1F3 committed:
Renamed the Southern Swamp flower rupees

G4M3RL1F3 committed:
Fixed maps switching to the wrong scenes

G4M3RL1F3 committed:
Fixed tracker not switching to "Termina" map for scenes that don't have a map yet

VincentsSin committed:
Location name cleanup and submap tracking

VincentsSin committed:
Added temporary maps for tracking

Southern Swamp, Snowhead, and Great Bay regions

G4M3RL1F3 committed:
Fix merge conflicts

G4M3RL1F3 committed:
Updated rules for groups

VincentsSin committed:
Updated json to display new maps

VincentsSin committed:
Delete duplicate Great Bay-Zora Hall tabs

G4M3RL1F3 committed:
Location sizes cleanup and put Termina Field map in Central tab

G4M3RL1F3 committed:
Fixed items not auto-tracking

G4M3RL1F3 committed:
Fixed sections with duplicate names, causing auto-tracking issues for Clock Tower

G4M3RL1F3 committed:
Fixed Clock Town Great Fairy Reward being invisible if setting is disabled

G4M3RL1F3 committed:
Fixed duplicate "North Clock Town" sections, causing auto-tracking issues

G4M3RL1F3 committed:
Fixed Mayor's Wife location not auto-tracking
Name mismatch.

G4M3RL1F3 committed:
Fixed missing Peahat Grotto section on Termina Field map

G4M3RL1F3 committed:
Fixed Kamaro locations not appearing on the Termina Field map

G4M3RL1F3 committed:
Fixed Guru-Guru location not auto-tracking
Name mismatch.

G4M3RL1F3 committed:
Fixed Road to Swamp grotto locations not auto-tracking
Name mismatches.

G4M3RL1F3 committed:
Fixed Woods of Mystery grotto locations not auto-tracking

G4M3RL1F3 committed:
Fixed duplicate "Southern Swamp" sections for Kotake Item and owl statue locations

G4M3RL1F3 committed:
Fixed Fisherman's grotto chest not auto-tracking

G4M3RL1F3 committed:
Removed fire arrows requirement from Song of Storms check

G4M3RL1F3 committed:
Fixed Road to Ikana grotto locations not auto-tracking

G4M3RL1F3 committed:
Cleaned-up Termina Field gossip stones grottos locations
I probably missed something, but it's too late for me to keep going. At least, it works for the heart piece check.

G4M3RL1F3 committed:
Fixed healing Mikau's check not auto-tracking

G4M3RL1F3 committed:
Separated which map purchases are physically available

G4M3RL1F3 committed:
Fixed Swamp Shooting Gallery auto-tracking

G4M3RL1F3 committed:
Fixed Priority Mail to Postman not auto-tracking

G4M3RL1F3 committed:
Fixed Monkey Song not auto-tracking and re-added its own map location

G4M3RL1F3 committed:
Fixed Deku Palace Bean Grotto locations not auto-tracking

G4M3RL1F3 committed:
Fixed Southern Swamp Grotto locations not auto-tracking

G4M3RL1F3 committed:
Fixed Swamp Spider House locations not auto-tracking

G4M3RL1F3 committed:
Fixed Deku Palace heart piece not auto-tracking

G4M3RL1F3 committed:
Fixed Woodfall region access rules being incorrect

G4M3RL1F3 committed:
Fixed Song of Soaring Tablet not auto-tracking

G4M3RL1F3 committed:
Fixed Curiosity Shop Backroom locations not auto-tracking

G4M3RL1F3 committed:
Fixed Woodfall Temple locations not auto-tracking

VincentsSin committed:
Fix tracking for Snowhead, Great Bay, and Stone Tower Temples

VincentsSin committed:
Fix Pirates' Fortress location tracking

G4M3RL1F3 committed:
Fixer Bomber's Hideout locations not auto-tracking

G4M3RL1F3 committed:
Fixed Baby Zora Song logic

G4M3RL1F3 committed:
Fixed Goron Village Deku Trade not auto-tracking

G4M3RL1F3 committed:
Fixed mountain region spring access rules

G4M3RL1F3 committed:
Fixed post-WFT "region" access rules

G4M3RL1F3 committed:
Fixed Termina Field Underwater Chest location not auto-tracking

G4M3RL1F3 committed:
Fixed Road to Ikana Pillar Chest not auto-tracking

G4M3RL1F3 committed:
Fixed South Clock Town chests not auto-tracking

G4M3RL1F3 committed:
Auto-tracking fixes and renames for swamp scrub

G4M3RL1F3 committed:
Fixed Butler Race location not auto-tracking

G4M3RL1F3 committed:
Fixed swamp Boat Archery not-autotracking

G4M3RL1F3 committed:
Zora Hall cleanup
- Every location was duplicated;
- Name changes;
- Auto-tracking issues.

G4M3RL1F3 committed:
Fixed Ocean Spider House chest not auto-tracking

G4M3RL1F3 committed:
Fixed Milk Bar Show not auto-tracking

G4M3RL1F3 committed:
Fixed Rosa Sisters locations not auto-tracking

G4M3RL1F3 committed:
Fixed Pirates' Fortress Sewers HP not auto-tracking

G4M3RL1F3 committed:
Fixed some Shiro locations not auto-tracking

G4M3RL1F3 committed:
Fixed some Ikana scrub locations not auto-tracking

G4M3RL1F3 committed:
Renamed scrub trades to "Deed Trade" (except Clock Town)

G4M3RL1F3 committed:
Fixed South Clock Town business scrub locations not auto-tracking

VincentsSin committed:
Fix tracking for various grass and other sanity locations

VincentsSin committed:
Fix tracking for deku flowers in Clock Town and Deku Palace

VincentsSin committed:
Added "beans" item to `can_plant_beans` macro

VincentsSin committed:
Revert bean logic to current apworld logic

G4M3RL1F3 committed:
Fixed Twin Islands Ramp Grotto Chest not auto-tracking

G4M3RL1F3 committed:
Fixed frog choir check not auto-tracking

G4M3RL1F3 committed:
Refined region clears logic

G4M3RL1F3 committed:
Added visibility rules to GBT Underneath Boss Door Platform Pots

G4M3RL1F3 committed:
Fixed GBT region clear rules

G4M3RL1F3 committed:
Fixed Great Bay Great Fairy Reward not auto-tracking

G4M3RL1F3 committed:
Fixed Great Bay Coast Fisherman Game not auto-tracking

G4M3RL1F3 committed:
Fixed Mountain Village (Spring) grotto chest not-auto-tracking

G4M3RL1F3 committed:
Fixed Ikana Castle HP not auto-tracking

G4M3RL1F3 committed:
Fixed Postman Counting not auto-tracking

G4M3RL1F3 committed:
Fixed Stone Tower Great Fairy Reward not auto-tracking

G4M3RL1F3 committed:
Fixed broken logic for Moon/Majora's Lair access

G4M3RL1F3 committed:
Added visibility rules for Goron Trial chests

G4M3RL1F3 committed:
Fixed Zora Trial HP not auto-tracking

G4M3RL1F3 committed:
Updated overworld map, split sections, and updated rules accordingly
Maps added by Vincent are mostly untouched. They will eventually get replaced by better quality ones, so I didn't want to spend time fixing them.

VincentsSin committed:
Fix Zora Hall location mismatches

G4M3RL1F3 committed:
Added Clock Town sections on the new overworld map

G4M3RL1F3 committed:
Changed Termina Field grass patches sections to diamonds

VincentsSin committed:
Update location marker shapes for grottos and sanity locations

VincentsSin committed:
Fix `Woods Of Mystery` overworld icon

G4M3RL1F3 committed:
Updated "grass_sanity" setting

G4M3RL1F3 committed:
Updated "pot_sanity" option

G4M3RL1F3 committed:
Fixed Termina Field Pillar Grotto Chest not auto-tracking
I don't know how I messed that one up lmao

G4M3RL1F3 committed:
Fixed Eastern Gossip Stones Grotto Grass not auto-tracking

G4M3RL1F3 committed:
Fixed incorrect logic for Pirates' Fortress Sewers

G4M3RL1F3 committed:
Added an exception to "has_absurd_soul" for "soul_bulldozer"
The "has_x_soul" functions return true if the corresponding option is disabled, aka not shuffle souls into the pool. Because the Soul of Bulldozer normally opens up the rest of Gorman Ranch, "has_absurd_soul" needs to return false to basically tell the tracker that Gorman Ranch can never open up if the setting is disabled.

G4M3RL1F3 committed:
Removed duplicate section from Woods of Mystery map

G4M3RL1F3 committed:
Fixed Termina Bombable Rock Grotto Beehive not auto-tracking

G4M3RL1F3 committed:
Fixed Bomber's Hideout Chest logic

G4M3RL1F3 committed:
Fixed OSH Main Basement Pots not auto-tracking

G4M3RL1F3 committed:
Fixed Dining Room Pots not auto-tracking

G4M3RL1F3 committed:
Fixed rest of OSH pots not auto-tracking

G4M3RL1F3 committed:
Fixed Outside Zora Hall access rules

G4M3RL1F3 committed:
Fixed Woodfall Bridge Chest logic
Although, the logic in the apworld is incorrect...

G4M3RL1F3 committed:
Fixed logic issues for Snowhead and Zora Cape owl statues and added can_warp_out

VincentsSin committed:
Split SHT 2F snowballs for cleaner tracking, adjusted SHT basement checks to proper locations on dungeon map

VincentsSin committed:
Fix tracking for Goron Village spring heart piece

VincentsSin committed:
Fix Pirates' Fortress sewer barrel rupees tracking

VincentsSin committed:
Fix `Path to Snowhead` scarecrows not tracking

VincentsSin committed:
Various fixes to locations not auto-tracking. `100% Completion Goal` untouched, likely to be removed.

VincentsSin committed:
Add Zora Hall souls to `Evan's Song` logic

VincentsSin committed:
Add grotto soul and hookshot to GBC grotto butterfly fairies logic.

VincentsSin committed:
Add `Swamp Tour Guide` to Deku Palace access for Pictograph requirement

VincentsSin committed:
Added `Soul of Evan` to match apworld logic; not a currently tracked item, added as comments for now

VincentsSin committed:
Fix Secret Shrine sanity locations not populating on dungeon map

VincentsSin committed:
Add grass soul for Keaton Quiz requirement
